core/metacling/src/TClingClassInfo.h

```diff
@@ -52,6 +53,8 @@ namespace ROOT {
 
 extern "C" typedef ptrdiff_t (*OffsetPtrFunc_t)(void*, bool);
 
+enum class ECtorCategory : short { kAbsent, kDefault, kTRootIOPtr, kTRootIORef, kVoidRef };
```

Comment:
What the difference between ECtorCategory and EIOCtorCategory ?  Do we really need the 2 enums?  If so, can we document the difference in semantic and usage (i.e. the why)?  Thanks.

Reply:
EIOCtorCategory is return value of `ROOT::TMetaUtils::CheckConstructor()` method.
It says, if constructor with provided type exists and has reference or pointer on specified type.

ECtorCategory is (for the moment) closed list of possible I/O constructor signatures.
Based on this value, one can generate helper function to produce wrapper for objects creation. 

One can made more generic class instead of simple `enum EIOCtorCategory` - then this class can be used in both cases.

Reply:
I think it would the most sense to go directly/now to the "more generic class" rather than having a kind of almost duplication.

Reply:
Ok, I made intermediate solution. Together with return value one can get constructor argument type name. In  this case `ECtorCategory` not necessary